### PR TITLE
Fixed #33113 -- Doc'd a forms.DateInput type=date example.

### PR DESCRIPTION
**Pull request by dennisvang**
This is my first pull request for django, so please let me know if anything could be improved.
Thanks.

ticket-33113

### CODE REVIEW

**Comment by claudep** [comment]
We could maybe refer to discussions on the ticket https://code.djangoproject.com/ticket/16630, or are they too old to be still valid?

File: docs/ref/forms/widgets.txt [hunk 1]
```diff
@@ -577,6 +577,23 @@ These widgets make use of the HTML elements ``input`` and ``textarea``.
...
+
+        class MyDateInput(forms.DateInput):
+            input_type = 'date'
+
+    Alternatively, you could specify the widget as follows: ::
+
+        DateInput(attrs={'type': 'date'})
+
+.. _HTML5 "date" input type: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date
+
```
Comment (carltongibson) [inline docs/ref/forms/widgets.txt:596]: I don't think we need the full `admonition` call out here. 

Just paragraphs should suffice:


```suggestion
    By default, :class:`~django.forms.DateInput` is rendered as an ``input``
    with ``type=text``. If you are targetting browsers which support the the
    `HTML5 "date" input type`_, you may override the ``input_type`` attribute,
    for example::
    
        class MyDateInput(forms.DateInput):
            input_type = 'date'

    Alternatively, you can specify the widget `attrs` directly::

        DateInput(attrs={'type': 'date'})

.. _HTML5 "date" input type: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date        
```

**Review by carltongibson — COMMENTED** [review]
Hi @dennisvang — I've suggested an edit... (@claudep's comment has crossed so please review.) 

I think we can call this `Fixed #... -- …` rather than just `Refs ... ` 😀

Thanks

**Comment by carltongibson** [comment]
@claudep — The reason **not** to use `type=date` is localisation issues, is that right? Do you have any thoughts for what you'd say here to that effect?

**Comment by claudep** [comment]
I guess that issues varies from browser to browser. It would be great to get feedback from someone having recently experienced with such use cases (especially if such issues remain). Sorry, I'm a bit short of inspiration right now for a good way of expressing our current choice.

File: docs/ref/forms/widgets.txt (hunk 1 above)
Comment (dennisvang) [reply docs/ref/forms/widgets.txt:596]: Thanks @carltongibson, that does look cleaner. Good point mentioning browser support. Personally I would prefer to mention "date picker" explicitly, somewhere in there: I expect quite a few users will be unaware of the fact that most browsers implement a fully-functional date picker interface for inputs with `type="date"`. At least it took *me* a while to find that out. ;-)

**Comment by carltongibson** [comment]
@claudep Let me ask on the mailing list. Thanks!

File: docs/ref/forms/widgets.txt (hunk 1 above)
Comment (carltongibson) [reply docs/ref/forms/widgets.txt:596]: @dennisvang — OK, have a stab — it's a _suggestion_. 😀

Comment (claudep) [reply docs/ref/forms/widgets.txt:596]: Well not completely convinced either. I think all browsers `support` the "date" input type, so I'd also rather mention the date picker at some point (`leverage the native browser's data pickers` ? or somehting like that).

**Comment by ngnpope** [comment]
> I think all browsers `support` the "date" input type, ...

> The reason not to use type=date is localisation issues, is that right?

Yes, we need to explain the reasoning. Otherwise the next person will come along and say, "But don't all browsers support `<input type="date">`? Why isn't this the default? Let's create a new ticket / pull request to 'fix' it...".

> It would be great to get feedback from someone having recently experienced with such use cases (especially if such issues remain).

Indeed, the first thing is to identify whether there is still an issue that blocks us from doing this by default. Maybe something has changed over the past decade...

----

As an aside, surely the localisation issue was also the reason for not including the following:

- [`datetime-local`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local) for `DateTimeField`. _(Interestingly, there was a [`datetime`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime), but it was deprecated.)_
- [`time`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/time) for `TimeField`.

Perhaps the documentation should also be updated for these too while we're at it?

**Comment by claudep** [comment]
Just a note: If at some point we think that `date` input types are now safe, we could start by dogfooding in the admin site (before changing the default widget).

**Review by blockchainGuru1018 — APPROVED** [review]

**Comment by nessita** [comment]
> I guess that issues varies from browser to browser. It would be great to get feedback from someone having recently experienced with such use cases (especially if such issues remain). Sorry, I'm a bit short of inspiration right now for a good way of expressing our current choice.

An example of a very very recent instance is [ticket 34853](https://code.djangoproject.com/ticket/34853)

The reporter had a couple of date fields (`borrowed_from_date`  and `borrowed_to_date`) in a model instance, that were rendered in a ModelForm as both localized **and** with an override to have them being:

```python
    widgets = {
        ...,
        "borrowed_from_date": forms.DateInput(attrs={"type": "date"}),
        "borrowed_to_date": forms.DateInput(attrs={"type": "date"}),
    }
```

The reporter's site also offered a dropdown to change the preferred language using the `set_language`  view. The main issue here is a disconnect between the language cookie and the browser's locale (sent via the `Accept-Language` header).

**Comment by nessita** [comment]
> Just a note: If at some point we think that `date` input types are now safe, we could start by dogfooding in the admin site (before changing the default widget).

I believe this is PR #15806?

**Comment by nessita** [comment]
> @claudep Let me ask on the mailing list. Thanks!

Discussion is https://groups.google.com/g/django-developers/c/wp-pnzcB25o/m/D5gEOzPIAQAJ

**Comment by nessita** [comment]
@dennisvang Do you have time to keep working on this?

**Comment by nessita** [comment]
Closing due to inactivity.

**Pull request closed without merging**